engine/execution/ingestion/core.go

```diff
@@ -88,20 +89,25 @@ func NewCore(
 		collectionFetcher: collectionFetcher,
 		eventConsumer:     eventConsumer,
 	}
+
+	e.ComponentManager = component.NewComponentManagerBuilder().
+		AddWorker(e.launchWorker).
+		Build()
+
+	return e
 }
 
-func (e *Core) Ready() <-chan struct{} {
+func (e *Core) launchWorker(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
 	if e.stopControl.IsExecutionStopped() {
-		return e.unit.Ready()
+		ready()
+		return
 	}
 
-	e.launchWorkerToConsumeThrottledBlocks()
+	e.ctx = ctx
```

Comment:
I think should be OK, because any function that uses `ctx` won't be called until `ready` is called. And `ready` is called after e.ctx is assigned, so should be ok.